### DISCUSSION TITLE
[Idea] Add an option in the cart rule to Apply discount on the original subtotal or not

### DISCUSSION THREAD

**Discussion by ioweb-gr** [PrestaShop/PrestaShop#32286] [Ideas]
**Improvement**
Apply a discount on the original subtotal is the choice we need. From what I see in other systems this is usually a dropdown which says

Apply on:
[Original Subtotal]
[Subtotal]

**Describe the bug**
I've created two rules that give a discount 8% and 2%. The amount in my cart is 57.35 so the discount should be 5.735  and the total should be calculated as 

57.35 - 5.735 = 51.615

On the contrary prestashop is doing the calculations on the already discounted price thus ending up with a smaller discount for the customer. 

So in succession it's calculating 
57.35 - 8% = 52.762 -2% = 51.70676 

There is no option to configure this on the backend whether you want to do it this way or the other way but the thing is if the discounts are not calculated based on the original subtotal it's going to be very confusing for the users. Because the administrator is going to have to consider rule priority in order to give the correct discount amount and the customers will definitely nag because they think the discount doesn't add up. 

**To Reproduce**
- create 2 discount rules for 8% and 2% excluding shipping
- create a product with value 57.35 
- go to cart and notice this cart result with the two discounts applied 

![image](https://user-images.githubusercontent.com/20220341/65612112-70c03080-dfbc-11e9-824f-0217cb05d804.png)

- Furthermore the discount amount is also incorrectly calculated. It should be the difference between the two amounts so at least it should say 
4.588 + 1.05524 = 5.64324  which is approximately 57.35 (subtotal) - 51.71 (total)

**Screenshots**
If applicable, add screenshots or screenrecords to help explain your problem.

**Additional information**
PrestaShop version:1.7.5.2 - 1.7.6.1
PHP version: 7.2

Recommended solution in my opinion: 
Add an option to cart rules "Apply to" with two options 
- Apply on subtotal 
- Apply on discounted subtotal 

In either case this scenario which is very common in B2B won't work as it is now. 

And in any case the discount amount should be calculated correctly. I can't even begin to understand which calculation shows 7 as discount amount. 

Kind regards

**Comment by ioweb-gr** [comment]
If anyone could point me to where I should begin to debug this I could probably offer more insight to why this is happening.

**Comment by khouloudbelguith** [comment]
Hi @ioweb-gr,

>There is no option to configure this on the backend whether you want to do it this way or the other way but the thing is if the discounts are not calculated based on the original subtotal it's going to be very confusing for the users

I think it could be a good improvement if we add this option in the cart rule conditions.
What do you think? it is an improvement & not a bug

Thanks!

**Comment by ioweb-gr** [comment]
Keep in mind the discount value is still calculated wrong and I think this type of calculation is important

**Comment by idnovate** [comment]
Related with https://github.com/PrestaShop/PrestaShop/issues/11722?

**Comment by ioweb-gr** [comment]
Reading upon it the symptoms are similar but in that case the OP made a separation for free gifts.  This is a plain discount rule.

**Comment by khouloudbelguith** [comment]
Hi @idnovate,

I think your issue is different.
Maybe related with this ticket:https://github.com/PrestaShop/PrestaShop/issues/11541
@ioweb-gr, We are aware of the Round issue PrestaShop as reported in this ticket: https://github.com/PrestaShop/PrestaShop/issues/9779

Thanks to check & feedback.

**Comment by ioweb-gr** [comment]
This cannot be a simple rounding issue.. We're talking about a difference of 1.5Euros in my example.

**Comment by khouloudbelguith** [comment]
@ioweb-gr, in my case, I did not manage to reproduce the issue with PS1.7.6.1.
The discount is well calculated = 5.64.
![image](https://user-images.githubusercontent.com/16067358/65675287-d4427080-e045-11e9-95d5-41d661c9c5db.png)
https://drive.google.com/file/d/18TZBlmr9OKYbvHkFNRiOuPq7mVw-1pqz/view

Thanks!

**Comment by ioweb-gr** [comment]
@khouloudbelguith Can you test with setting prices to excl. tax

**Comment by khouloudbelguith** [comment]
@ioweb-gr, what is the exact tax rule did you use for your product?
![image](https://user-images.githubusercontent.com/16067358/65676135-45365800-e047-11e9-9e9a-0c81df1893c5.png)
Thanks!

**Comment by ioweb-gr** [comment]
OK I re-setup a 1.7.6.1 instance and the discount calculation issue seems to have been fixed.

**Comment by marionf** [comment]
@ioweb-gr This issue can be closed ?

**Comment by ioweb-gr** [comment]
Well the original problem that multiple discounts cannot be applied on subtotal but only consecutively on the discounted price remains. Perhaps rename the title to better reflect it and leave it open as an important feature request?

**Comment by idnovate** [comment]
But this is not a bug but a request ;)

**Comment by ioweb-gr** [comment]
I agree this Depends on whether the team wanted it to work like this or not and the merchant's point of view  :) in my eyes it's an issue because the customers will feel that the merchant is scamming them even though they might be wrong. 

I asked about this scenario 5 random people at a cafe just to get a rough idea what people would think. They all agreed 8% and 2% means 10% hehe. I bet merchants can expect a lot of phone calls if there's no option to modify how it works

**Comment by idnovate** [comment]
> I bet merchants can expect a lot of phone calls if there's no option to modify how it works

It has been working this way since the beginning and I don't know of any death until now :v:
But not saying that it should work as you describe ;)

**Comment by khouloudbelguith** [comment]
@ioweb-gr, so as a conclusion.
We need to edit this issue from a bug to an improvement.
Your exact improvement is to have an option in the cart rule configuration "Apply the discount on the original subtotal" Yes or NO.

>There is no option to configure this on the backend whether you want to do it this way or the other way but the thing is if the discounts are not calculated based on the original subtotal it's going to be very confusing for the users

Can you please confirm this?

Thanks!

**Comment by ioweb-gr** [comment]
Yes it sounds about right. Apply discount on the original subtotal is the choice we need. From what I see in other systems this is usually a dropdown which says 

Apply on:
[Original Subtotal]
[Subtotal]

So later on other options can be added as well by modules extending the functionality.

**Comment by khouloudbelguith** [comment]
Ping @PrestaShop/prestashop-product-team what do you think?

Thanks!
